### PR TITLE
LATIS-1061 munit migration

### PR DESCRIPTION
**Pull request by JakeMartin99**
In progress... upgrading tests to munit

### CODE REVIEW

**Comment by JakeMartin99** [comment]
Ok, should be good for review now

File: src/test/scala/latis/service/hapi/TimeParseSuite.scala [hunk 1]
```diff
@@ -2,57 +2,54 @@ package latis.service.hapi
...
+  val expected: LocalDateTime = LocalDateTime.of(2018, 1, 1, 0, 0, 0)
 
-  def testParse(str: String): Assertion =
-    assert {
-      expected === Time.parse(str).getOrElse {
-        fail(s"Failed to parse time string: $str")
-      }
+  def testParse(str: String): Unit =
+    Time.parse(str) match {
+      case Some(t) => assert(t === expected)
```
Comment (dlindhol) [inline src/test/scala/latis/service/hapi/TimeParseSuite.scala:15]: Should we prefer `assertEquals`?

File: src/test/scala/latis/service/hapi/BinSuite.scala [hunk 2]
```diff
@@ -2,11 +2,11 @@ package latis.service.hapi
 
 import io.circe._
 import io.circe.syntax._
-import org.scalatest.flatspec._
+import munit.CatsEffectSuite
 
-class BinSpec extends AnyFlatSpec {
+class BinSuite extends CatsEffectSuite {
```
[the remark this answers is not in the document]
Comment (JakeMartin99) [reply src/test/scala/latis/service/hapi/BinSuite.scala:7]: Certainly possible. I was just in a "`CatsEffectSuite`" mindset and converted everything the same way for consistency.

File: src/test/scala/latis/service/hapi/TimeParseSuite.scala (hunk 1 above)
Comment (JakeMartin99) [reply src/test/scala/latis/service/hapi/TimeParseSuite.scala:15]: From the existing comment on the `implicit val` that implements `===`:
  // The `equals` method should be used to compare `LocalDateTime`
  // instances. See the Javadocs.
So I was wary to change it.

Comment (JakeMartin99) [reply src/test/scala/latis/service/hapi/TimeParseSuite.scala:15]: I suppose it depends on whether `assertEquals` is doing a `==` or a `.equals` behind the scenes...

File: src/test/scala/latis/service/hapi/BinSuite.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (JakeMartin99) [reply src/test/scala/latis/service/hapi/BinSuite.scala:7]: Seems to work fine

File: src/test/scala/latis/service/hapi/TimeParseSuite.scala (hunk 1 above)
Comment (JakeMartin99) [reply src/test/scala/latis/service/hapi/TimeParseSuite.scala:15]: Seems to work fine

File: src/test/scala/latis/service/hapi/DataServiceSuite.scala [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+      Sample(DomainData("2000-01-04"), RangeData(2)),
+      Sample(DomainData("2000-01-05"), RangeData(0)),
+    ))
+    dataset = new MemoizedDataset(Metadata("id"->"testdataset"), model, data)
+  } yield dataset).fold(err => fail(err.message), identity)
+  private lazy val latisInterp = new Latis3Interpreter(Catalog(dataset))
+  private lazy val dataService = new DataService[IO](latisInterp).service
+
+  /** Assert GET request to given URI returns a particular status. */
+  def assertStatus(uri: Uri, status: HStatus): IO[Unit] = {
```
Comment (dlindhol) [inline src/test/scala/latis/service/hapi/DataServiceSuite.scala:49]: Should we include the implicit Location?
https://scalameta.org/munit/docs/tests.html#declare-tests-inside-a-helper-function

Comment (JakeMartin99) [reply src/test/scala/latis/service/hapi/DataServiceSuite.scala:49]: Sure, might be useful

File: src/test/scala/latis/service/hapi/DataServiceSuite.scala [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+      HStatus.`1402`
+    )
+    assertStatus(
+      uri"/data?dataset=0&start=invalid&stop=2018Z",
+      HStatus.`1402`
+    )
+    assertStatus(
+      uri"/data?dataset=0&start=invalid&stop=2018Z",
+      HStatus.`1402`
+    )
```
[the remark this answers is not in the document]
Comment (JakeMartin99) [reply src/test/scala/latis/service/hapi/DataServiceSuite.scala:77]: They look the same to me too

**Review by lindholc — APPROVED** [review]

**Review by dlindhol — APPROVED** [review]

**Pull request merged**